__tests__/app.test.js

```diff
@@ -73,6 +73,37 @@ describe("/api/articles", () => {
           });
         });
     });
+
+    test("GET 200: Accepts a topic query and responds with an array of articles of that topic", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.articles.length).toBe(12);
+
+          body.articles.forEach((article) => {
+            expect(article.topic).toBe("mitch");
+          });
+        });
+    });
+
+    test("GET 404: Responds with an adequate error when the topic does not exist", () => {
+      return request(app)
+        .get("/api/articles?topic=mocha")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("Not found");
+        });
+    });
+
+    test("GET 404: Responds with an adequate error when the column name does not exist", () => {
+      return request(app)
+        .get("/api/articles?inexistent=mitch")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.message).toBe("Not found");
+        });
+    });
```

Comment:
Good tests, remember though:

404 - Doesn't currently exist in the database but could

400 - Will never exist in the database without changing the columns

For line 99, considering you know all the valid columns to _ever_ filter by , there is no way you could filter by something else, without changing the db, so it would be a 400